### PR TITLE
CCM-4892: Create NHS App template page

### PR DESCRIPTION
**Pull request by chris-elliott-nhsd**
## Description

PR to add a create NHS App template page

Deployed environment https://feature-ccm-4892-create-nhsapp-template.d1ie8muegpw33w.amplifyapp.com/choose-template

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/choose-template/main-server-action.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    zodValidationServerAction(
+      formState,
+      formData,
+      z.object({
+        nhsAppTemplateName: z
+          .string()
+          .min(1, { message: 'Enter a template name' }),
+        nhsAppTemplateMessage: z
+          .string()
+          .min(1, { message: 'Enter a template message' }),
```
Comment (bhansell1) [inline src/app/choose-template/main-server-action.ts:46]: This should also have a max set to 5000. We can't rely on the client-side  input limiting the value to 5000. You can bypass the limit on the client by manually editing the HTML.

Comment (chris-elliott-nhsd) [reply src/app/choose-template/main-server-action.ts]: updated

**Review by bhansell1 — DISMISSED** [review]

File: src/app/create-template/page.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const CreateTemplate = () => {
+  const [state, action] = useFormState(mainServerAction, initialState);
+  const { page } = state;
+
+  const PageComponent = pages[page];
+
+  return (
+    <>
+      <PageComponent state={state} action={action} />
+    </>
```
[the remark this answers is not in the document]
Comment (chris-elliott-nhsd) [reply src/app/create-template/page.tsx]: updated

File: src/components/forms/CreateEmailTemplate/CreateEmailTemplate.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
-export default function CreateEmailTemplate() {
+export const CreateEmailTemplate = () => {
   return (
     <>
       <h2 className='nhsuk-heading-l' data-testid='page-sub-heading'>
         Placeholder email page
       </h2>
     </>
```
[the remark this answers is not in the document]
Comment (chris-elliott-nhsd) [reply src/components/forms/CreateEmailTemplate/CreateEmailTemplate.tsx]: updated

**Review by m-salaudeen — APPROVED** [review]

**Review by bhansell1 — DISMISSED** [review]

**Review by m-salaudeen — APPROVED** [review]

**Review by m-salaudeen — APPROVED** [review]

**Pull request merged**